engine/execution/ingestion/engine.go

```diff
@@ -471,29 +464,11 @@ func (e *Engine) executeBlock(
 		return
 	}
 
-	// if the receipt is for a sealed block, then no need to broadcast it.
-	lastSealed, err := e.state.Sealed().Head()
-	if err != nil {
-		lg.Fatal().Err(err).Msg("could not get sealed block before broadcasting")
-	}
-
 	receipt := computationResult.ExecutionReceipt
-	isExecutedBlockSealed := executableBlock.Block.Header.Height <= lastSealed.Height
-	broadcasted := false
-
-	if !isExecutedBlockSealed {
-		authorizedAtBlock, err := e.checkAuthorizedAtBlock(executableBlock.ID())
-		if err != nil {
-			lg.Fatal().Err(err).Msg("could not check staking status")
-		}
-		if authorizedAtBlock {
-			err = e.providerEngine.BroadcastExecutionReceipt(ctx, receipt)
-			if err != nil {
-				lg.Err(err).Msg("critical: failed to broadcast the receipt")
-			} else {
-				broadcasted = true
-			}
-		}
+	broadcasted, err := e.providerEngine.BroadcastExecutionReceipt(
```

Comment:
this looks like the only use of the returned boolean, and that is only for logging. Does it still make sense to return this boolean? If we still want this in the log, can we  not move the logging inside of `BroadcastExecutionReceipt`

Reply:
Yes, it's only used for logging for now. And yes, we could move the logging to the provider engine, but it's not hurt to still return the boolean. I'd like to keep as is so that we don't have to introduce additional logging just to know whether it's broadcasted. It's simple to just merge the broadcaster along with other property of the receipts, so that it's easy to search for the receipt.